### PR TITLE
Use rollup instead of Gulp

### PR DESCRIPTION
**Pull request by azhalkouski**
https://github.com/McReader/observe-props-decorator/issues/5

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import resolve from 'rollup-plugin-node-resolve';
```
Comment (McReader) [inline rollup.config.js]: format code, please

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    format: 'cjs',
+    name: 'ObservePropsDecorator',
+		sourcemap: true
+	},
+	plugins: [
+		resolve({
+      customResolveOptions: {
+        moduleDirectory: 'node_modules'
+      }
+    }), // tells Rollup how to find date-fns in node_modules
```
Comment (McReader) [inline rollup.config.js]: date-fns?

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    commonjs({ // converts date-fns to ES modules
```
Comment (McReader) [inline rollup.config.js]: why do we need commonjs?

File: src/observeProps.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React, { PureComponent } from 'react';
 
-import { Subject } from 'rxjs/internal/Subject';
-import { from } from 'rxjs/internal/observable/from';
+import { Subject } from 'rxjs';
```
Comment (McReader) [inline src/observeProps.js]: Why?

File: rollup.config.js (hunk 1 above)
Comment (azhalkouski) [reply rollup.config.js]: Sorry, I missed it.

File: rollup.config.js (hunk 2 above)
Comment (azhalkouski) [reply rollup.config.js]: removed

File: rollup.config.js (hunk 3 above)
Comment (azhalkouski) [reply rollup.config.js]: We need CommonJS plugin to convert CommonJS modules to ES6, so they can be included in a Rollup bundle.
Without the plugin:
1) rxjs imports and lodash with exact paths aren't handled by Rollup;
2) the final bundle's size is by 3 times bigger than with the plugin.

Thus, the plugin fixes imports and helps with bundle size optimization. I suppose that rxjs, lodash and more other libs are available for use like cjs modules.

**Pull request merged**